### ISSUE TITLE
Translation (DE) does not use uppercase on Competitions page

### ISSUE THREAD

**Issue by AnnikaStein** [thewca/worldcubeassociation.org#7299]
**Describe the bug**
The keys `show_cancelled` as well as `show_registration_status` did not pick up the translation with uppercase notation, despite it being written correctly in the locales file (see here: https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/config/locales/de.yml#L2984-L2987). This had been merged a month ago, and another part of that PR is correctly visualized, so I think the build has correctly used the most recent .yml file, it's just these two keys not acting as they should.

**To Reproduce**
1. Go to https://www.worldcubeassociation.org/competitions
2. Switch language to "Deutsch" (= German)

**Expected behavior**
Translation should maintain lower-/uppercase exactly as written in locales file

**Screenshots**
<img width="1653" alt="Bildschirmfoto 2022-10-15 um 22 15 05" src="https://user-images.githubusercontent.com/53974095/196006885-01876ba8-2a76-4fbb-9c89-a00d87651c14.png">

**Desktop (please complete the following information):**
 - OS: MacOS 12.3.1
 - Browser: Chrome
 - Version: 106.0.5249.119 (Official Build) (arm64)

**Smartphone (please complete the following information):**
 - Device: iPhone 12 mini
 - OS: iOS 14.2.1
 - Browser: safari

**Additional context**
- Happens only for the German translation, which is (besides Letzebuergesch maybe) the only language I know for which the uppercase does indeed matter in this particular context.
- I could exclude that the newest YAML file might not have been used in the latest website build, and base this statement on the observation that a new translation which was added in the same PR successfully made it to e.g. the user edit page (the `registration_notifications_enabled` key was not there before, and the correct translation appears in my personal profile next to the checkbox).
- I checked a couple of other languages, the translations as shown on the website seemed to match what translators wrote into the respective YAMLs.
- Signing in/out has no effect on the problem.
- Problem also occurs when testing locally (using the docker container setup).
- I tested putting the translation into quotation marks (single or double), it did not change the situation. If I prepend the "Zeige" with something, also "Zeige" will not be shown with capital Z. From this I conclude there is something that automatically renders every character after the first one with lowercase letters.
- Proposed fix: If I don't use the `label_tag` that currently surrounds the text (see https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/views/competitions/_index_form.html.erb#L116-L124), it works. But I'm not sure if this is how it should be. If it's fine like that, I can create a PR for this and / or work out a better solution. Below you can see a comparison with / without the `label_tag` and how the result looks (just using a dummy translation text for demo purposes):
locales/de.yml looks like that
<img width="978" alt="Bildschirmfoto 2022-10-15 um 23 14 04" src="https://user-images.githubusercontent.com/53974095/196007838-c52ee538-1a34-42cb-9478-10d22c1e0744.png">
Changes for worldcubeassociation.org/WcaOnRails/app/views/competitions/_index_form.html.erb (note that I would also change the first occurrence of the label_tag issue in the final version, left it in here for demonstration only)
<img width="1052" alt="Bildschirmfoto 2022-10-15 um 23 14 28" src="https://user-images.githubusercontent.com/53974095/196007846-89bfa4a9-555c-4190-825b-1537631a3916.png">
And the result:
<img width="1197" alt="Bildschirmfoto 2022-10-15 um 23 12 07" src="https://user-images.githubusercontent.com/53974095/196007746-d453ce35-656c-4289-b803-cc3a011680ca.png">
The behaviour for e.g. locales/en is unchanged. Neither errors nor warnings introduced.

TL;DR: without the `label_tag` the text is "translated" correctly, otherwise, everything is lowercase.

**Comment by dunkOnIT** [comment]
Hi there Annika! Thanks very much for the detailed write-up, and the proposed solution. A PR for it would be very appreciated - @gregorbg would you have capacity to take a brief look at the proposed solution (last bullet onwards) and advise if Annika can go ahead with an implementation of it? Or are there any issues that would create that we're not aware of?

**Comment by AnnikaStein** [comment]
@dunkOnIT @gregorbg

Some additional context, as I wasn't completely happy with removing the tag:
I think I now found out what would be a proper way to describe the observation in terms of ruby: it's the [`humanize` method](https://apidock.com/rails/String/humanize) which capitalizes the first word, but not the rest, humanize is called when using [`label_tag`](https://apidock.com/rails/ActionView/Helpers/FormTagHelper/label_tag) without specifying the content explicitly, which is exactly what happens in the original code snippet posted above.

This means a second viable way to handle the issue would be to pass the i18n key twice inside `label_tag`, once for `name` and once for `content_or_options`. Then we should be able to keep the desired structural element <label etc.> inside the DOM, and at the same time get the text or content exactly as stored in the .yml file.

**Comment by dunkOnIT** [comment]
Hi @AnnikaStein - thanks for the extra research there. 

It sounds like you've gotten to the root of the issue, and the new solution sounds much more elegant than just removing `label_tag` - which also "smelled" a bit wrong to me :P 

I would say you can/should go ahead with the PR for passing the i18n key for `content_or_options` as well without the need for Gregor's input.
